### DISCUSSION TITLE
volume defined as folder in S3 bucket cannot be mounted

### DISCUSSION THREAD

**Discussion by karapuz50** [datashim-io/datashim#343] [Q&A]
Hi

I'm running with datashim-charts-0.4.0 and trying to define a volume as a folder from specific S3 bucket. 
dataset object is:
Name:         s3-jfile
Namespace:    dmitryp
Labels:       <none>
Annotations:  API Version:  datashim.io/v1alpha1
Kind:         Dataset
Metadata:
  Creation Timestamp:  2024-03-20T08:15:14Z
  Generation:          1
  < truncated> 
    Manager:         manager
    Operation:       Update
    Subresource:     status
    Time:            2024-03-20T08:15:19Z
  Resource Version:  61384766
  UID:               f2f9970f-92a9-41eb-af26-8904c85e3b70
Spec:
  Local:
    Bucket:              jfile-eks-devops-us-east-2/jfile
    Endpoint:            https://s3.us-east-1.amazonaws.com
    Provision:           true
    Readonly:            false
    Region:              us-east-1
    Secret - Name:       s3secret
    Secret - Namespace:  dmitryp
    Type:                COS
Status:
  Caching:
    Info:    No DLF caching plugins are installed
    Status:  Disabled
  Provision:
    Status:  OK
Events:      <none>

Here is the log from csi-s3 DS running on the node where volume was mounted:

0320 08:24:48.044063       1 nodeserver.go:75] target /var/lib/kubelet/pods/43464474-fdab-480c-bdab-0764b7b89568/volumes/kubernetes.io~csi/pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5/mount
device 
readonly false
volumeId pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5
attributes map[mounter:goofys storage.kubernetes.io/csiProvisionerIdentity:1710922176654-8081-ch.ctrox.csi.s3-driver]
mountflags []
I0320 08:24:48.044179       1 mounter_goofys.go:26] Mounting using goofys volume pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5
I0320 08:24:48.044191       1 mounter_goofys.go:53] Mounting using goofys!
I0320 08:24:48.044241       1 mounter.go:56] Mounting fuse with command: goofys and args: [--endpoint=https://s3.us-east-1.amazonaws.com --type-cache-ttl 1s --stat-cache-ttl 1s --dir-mode 0777 --file-mode 0777 --http-timeout 5m -o allow_other --profile=pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5 --region us-east-1 jfile-eks-devops-us-east-2 /var/lib/kubelet/pods/43464474-fdab-480c-bdab-0764b7b89568/volumes/kubernetes.io~csi/pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5/mount]
I0320 08:24:49.105045       1 nodeserver.go:144] s3: bucket jfile-eks-devops-us-east-2 successfuly mounted to /var/lib/kubelet/pods/43464474-fdab-480c-bdab-0764b7b89568/volumes/kubernetes.io~csi/pvc-1eac2e9a-7070-42df-825e-d2013e80e5e5/mount
I0320 08:25:35.801875       1 utils.go:97] GRPC call: /csi.v1.Node/NodeGetCapabilities
I0320 08:25:44.519060       1 utils.go:97] GRPC call: /csi.v1.Node/NodeGetCapabilities

here is screenshot of the s3 bucket's content:
![s3-mount](https://github.com/datashim-io/datashim/assets/29249258/cfb19a37-2a98-4c9a-97b4-789a62ec0a3d)

as you see there is no folder 'jfile' inside and all files are created in the root of the bucket.

What I'm doing wrong? Please help.

Dmitry

**Comment by karapuz50** [comment]
adding here log from dataset-operator during the dataset provisioning.

.7109429499856966e+09	INFO	controller_datasetinternal	All good, proceed	{"Request.Namespace": "dmitryp", "Request.Name": "s3-jfile"}
1.7109429499857087e+09	INFO	controller_datasetinternal	Secret namespace is same as dataset namespace, allowed	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "checkAndExtractObjectStorageAccess", "Dataset.Name": "s3-jfile"}
1.7109429499857254e+09	INFO	controller_datasetinternal	Authentication info has been successfully retrieved	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset", "Dataset.Name": "s3-jfile"}
1.7109429499857342e+09	INFO	controller_datasetinternal	Found a prefix associated with the bucket for Dataset	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset", "Dataset.Name": "s3-jfile", "bucket": "jfile-eks-devops-us-east-2", "prefix": "jfile"}
1.7109429499857397e+09	INFO	controller_datasetinternal	Region extracted from Dataset	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset", "Dataset.Name": "s3-jfile", "bucket": "jfile-eks-devops-us-east-2", "region": "us-east-1"}
1.7109429499857657e+09	INFO	controller_datasetinternal	Creating new configMap	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "createConfigMapforObjectStorage", "configMap.namespace": "dmitryp", "configMap.Name": "s3-jfile"}
1.710942949993831e+09	INFO	controller_datasetinternal	Creating new pvc	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset", "PVC.Namespace": "dmitryp", "PVC.Name": "s3-jfile"}
1.7109429500089216e+09	INFO	controller_datasetinternal	Creating new secrets	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset", "Secret.Namespace": "dmitryp", "Secret.Name": "s3-jfile"}
1.710942950015455e+09	INFO	controller_datasetinternal	All good and we shouldnt reconcile!	{"Dataset.Namespace": "dmitryp", "Dataset.Name": "s3-jfile", "Method": "processLocalDataset"}
1.710942954985516e+09	INFO	controller.dataset	Reconciling Dataset	{"reconciler group": "datashim.io", "reconciler kind": "Dataset", "name": "s3-jfile", "namespace": "dmitryp", "Request.Namespace": "dmitryp", "Request.Name": "s3-jfile"}
1.710942954994178e+09	INFO	controller.dataset	Reconciling Dataset	{"reconciler group": "datashim.io", "reconciler kind": "Dataset", "name": "s3-jfile", "namespace": "dmitryp", "Request.Namespace": "dmitryp", "Request.Name": "s3-jfile"}
1.7109429549942133e+09	INFO	controller.dataset	Dataset Internal Instance exists, check if different EDITABLE labels and update	{"reconciler group": "datashim.io", "reconciler kind": "Dataset", "name": "s3-jfile", "namespace": "dmitryp", "Request.Namespace": "dmitryp", "Request.Name": "s3-jfile"}

**Comment by karapuz50** [comment]
can be closed. worked as expected after another test

**Comment by srikumar003** [comment]
@karapuz50 apologies on missing out on this question. Was there anything you changed to make it work, or was it just a retry of the same test ? Your logs don't show any errors, so I am curious how the issue was resolved.

(Please open an issue, if you come across more problems, as we don't look at the Discussion section for now)
